fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckService.kt

```diff
@@ -100,7 +100,7 @@ class DefaultReviewQualityCheckService(
                 tab.engineState.engineSession?.requestAnalysisStatus(
                     url = tab.content.url,
                     onResult = {
-                        continuation.resume(it.asEnumOrDefault<AnalysisStatusDto>())
+                        continuation.resume(it.asEnumOrDefault(AnalysisStatusDto.OTHER))
```

Comment:
non-blocking:
now that `asEnumOrDefault` is being passed a default value, should the function be simplified and made non-null and simply hardcode `OTHER` as the fallback?

Reply:
Yeah that would make sense for `ReviewQualityCheckService`, but I think we should move this to common/middleware package as  product recommendations can use the same functionality and the existing String.toGrade mapping in `ProductAnalysisMapper.kt` can be replace by the common function, wdyt?

Reply:
Makes sense to me